### PR TITLE
fix: expose server zone enum

### PR DESCRIPTION
**Pull request by kevinpagtakhan**
### Summary

Bring back ServerZone enum for ampli compatibility

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-TypeScript/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  No

### CODE REVIEW

File: packages/analytics-types/src/server-zone.ts [hunk 1]
```diff
@@ -1 +1,6 @@
+export enum ServerZone {
```
Comment (justin-fiedler) [inline packages/analytics-types/src/server-zone.ts:1]:
```suggestion
/**
 * @deprecated use ServerZoneType instead
 */
export enum ServerZone {
```

**Review by justin-fiedler — APPROVED** [review]
Looks good. Thanks @kevinpagtakhan. I think we should mark the enum deprecated to steer people to the union type.

**Review by falconandy — COMMENTED** [review]
@justin-fiedler @kevinpagtakhan 

> for ampli compatibility

1. What compatibility case should be supported?
2. Is it the only incompatibility?
3. Will new codegen `v3` be added to support new `2.x` SDK version? If yes, will codegen use new `ServerZoneType` type (actually constants `US`/`EU`)?

**Comment by kevinpagtakhan** [comment]
> @justin-fiedler @kevinpagtakhan
> 
> > for ampli compatibility
> 
> 1. What compatibility case should be supported?
> 2. Is it the only incompatibility?
> 3. Will new codegen `v3` be added to support new `2.x` SDK version? If yes, will codegen use new `ServerZoneType` type (actually constants `US`/`EU`)?

Hi @falconandy! I removed customer facing enums which includes `ServerZone` for Browser SDK 2.0. I then noticed that ServerZone is used by ampli v2. @justin-fiedler and I agreed that we should make Browser SDK 2.0 compatible with ampli v2, so it does not force us to work on an ampli v2. The work here is to bring back `ServerZone` enum but only for ampli v2 purposes.

**Pull request merged**